### DISCUSSION TITLE
ICA-based clustering question

### DISCUSSION THREAD

**Discussion by Dillon214** [satijalab/seurat#6639] [Q&A]
Hello Seurat team,

I have a question regarding independent component analysis, and dealing with variation caused by technical artifacts. 
I am a new bioinformatician and the goal for the analysis I'm working on is to analyze the differences between cells harvested from different tissues, however I am facing issues because some tissues were digested enzymatically, and others were digested mechanically, thus mixing biological signal with technical signal.
I found that RunICA did a good job of capturing the technical signal, and am interested in clustering with problematic ICs not included, but am unsure how many total ICs to use for clustering. Jackstrawplot and elbowplots throw an error.

Furthermore, I am interested in reducing the effect this technical artifact has on dowstream analysis, mainly DE and pathway analysis. I think it would be possible to use the feature loading scores for the offending ICs to determine a list of genes which are suspect, but I am not sure if this is the best approach. Any advice would be highly appreciated.

**Comment by alikhuseynov** [comment]
Hi there, 
I'm not from the Seurat team, however you can try independent PCA - [ipca](https://rdrr.io/bioc/mixOmics/man/ipca.html) from [mixOmics](http://mixomics.org/2012/04/sipca/)
which has both ICA and PCA fusion. 
Then you can add that to your seurat obj as another dim https://satijalab.org/seurat/archive/v3.0/dim_reduction_vignette.html
hope this helps.
